File: Bayes/tickinspec.r
#tickinspec.r
rm(list=ls());
#dev.off();
source("kde.r");
source("gamrand.r");
c = 3.481048347*10^(19);
n = 10000;
x = 60;
p = cbind(1/3,1/3,1/3);
kk = matrix(0,1,n);
tt = matrix(0,1,n);
k = min(which(cumsum(p)> runif(1)));
for (i in 1:n){
  a = x+ 1;
  b = 2/k + 10*k; 
  t = gamrand(a,b);
  tt[i] = t;
  p = cbind(exp(-12*t),2^(x-1)*exp(-21*t), 3^(x-1)*exp(-92*t/3));
  p = p/sum(p);
  k = min(which(cumsum(p)> runif(1)));
  kk[i] = k;  
}
p1est = sum(kk == 1)/n  #estimate of posterior probability 1
p2est = sum(kk == 2)/n  #estimate of posterior probability 2
p3est = sum(kk == 3)/n  #estimate of posterior probability 3
p1est
p2est
p3est
result<-kde(tt,len=0,loop=1);
h = seq(0,4,0.01);
q = h^(60)*(3^(59)*exp(-92*h/3) + 2^59*exp(-21*h) + exp(-12*h))/c;
lines(h,q,lwd=2);
#plot(h,q,xlim=c(1,4),ylim=c(0,1.2),lwd=2,type='l',xlab=" t", ylab='f(t|x) = 60');

File: Bayes/tickinspec.m
%tickinspec.m
clear all,clf
c = 3.481048347*10^(19);
t = 0:0.01:4;
f = t.^(60).*(3^(59)*exp(-92*t/3) + 2^59*exp(-21*t) + exp(-12*t))/c;
hold on
p = plot(t,f,'k','LineWidth',2);
% Uncomment the following line to preserve the X-limits of the axes
%  xlim([1 4]);
% % Uncomment the following line to preserve the Y-limits of the axes
%  ylim([0 1.2]);
% hold('all');
% 
% % Create plot
% plot(t,f,'LineWidth',2,'Color',[0 0 0]);
% 
% % Create xlabel
% xlabel('$t$','Interpreter','latex','FontSize',20);
% 
% % Create ylabel
% ylabel('$f(t \, |\, x = 60)$','Interpreter','latex','FontSize',20);
%     

n = 10000;
x = 60; %number of tickets
p = [1/3,1/3,1/3] %initial value
kk = zeros(1,n);
tt = zeros(1,n);
k = min(find(cumsum(p)> rand));
for i=1:n
    a = x+ 1;
    b = 2/k + 10*k; 
    t = gamrand(a,b);
    tt(i) = t;
    p = [exp(-12*t),2^(x-1)*exp(-21*t), 3^(x-1)*exp(-92*t/3)];
    p = p/sum(p);
    k = min(find(cumsum(p)> rand));
    kk(i) = k;
end
p1est = sum(kk == 1)/n  %estimate of posterior probability 1
p2est = sum(kk == 2)/n  %estimate of posterior probability 2
p3est = sum(kk == 3)/n  %estimate of posterior probability 3
kde(tt)
